File: AddressBook/react-client/src/dtos/Person.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";
import { PersonAddressRes } from "./PersonAddress";
import { PersonChatRes } from "./PersonChat";
import { PersonEmailRes } from "./PersonEmail";
import { PersonLabelRes } from "./PersonLabel";
import { PersonPhoneRes } from "./PersonPhone";
import { PersonWebsiteRes } from "./PersonWebsite";

export interface PersonRes {
  personId?: number;
  firstName?: string;
  middleName?: string;
  lastName?: string;
  pictureUrl?: string;
  company?: string;
  jobTitle?: string;
  department?: string;
  dateOfBirth?: Date;
  notes?: string;

  personLabels?: PersonLabelRes[];
  personEmails?: PersonEmailRes[];
  personPhones?: PersonPhoneRes[];
  personAddresses?: PersonAddressRes[];
  personWebsites?: PersonWebsiteRes[];
  personChats?: PersonChatRes[];
}

export class PersonEditReq {
  firstName?: string = "";
  middleName?: string = "";
  lastName?: string = "";
  pictureUrl?: string = "";
  company?: string = "";
  jobTitle?: string = "";
  department?: string = "";
  dateOfBirth?: Date = new Date();
  notes?: string = "";
}

export class PersonReqSearch extends PagedReq {
  personLabelId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId: personLabelId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personLabelId = personLabelId;
  }
}

File: AddressBook/AddressBook/Dtos/Person.cs
using AddressBook.Common.Paging;
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class PersonRes
    {
        public int PersonId { get; set; }
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public string? PictureUrl { get; set; }
        public string? Company { get; set; }
        public string? JobTitle { get; set; }
        public string? Department { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string? Notes { get; set; }

        // Child tables
        public IEnumerable<PersonLabelRes>? PersonLabels { get; set; }
        public IEnumerable<PersonEmailRes>? PersonEmails { get; set; }
        public IEnumerable<PersonPhoneRes>? PersonPhones { get; set; }
        public IEnumerable<PersonAddressRes>? PersonAddresses { get; set; }
        public IEnumerable<PersonWebsiteRes>? PersonWebsites { get; set; }
        public IEnumerable<PersonChatRes>? PersonChats { get; set; }
    }

    public class PersonReqEdit
    {
        [Required]
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public string? PictureUrl { get; set; }
        public string? Company { get; set; }
        public string? JobTitle { get; set; }
        public string? Department { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string? Notes { get; set; }
    }

    public class PersonReqSearch : PagedReq
    {
        public int? PersonLabelId { get; set; }
    }
}
